### PR TITLE
Arm mitigation

### PR DESCRIPTION
**Pull request by elleryjh**
- Create an alternate way to calibrate when we know we're already at the home position
- Add a timeout to retracting the arm in auto mode - this will allow us to reach set points if the motor is not strong enough to pull the arm in.

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -313,6 +314,7 @@ private void configureButtonBindings() {
 
     // Need to set to use automated movements, should be set in Autonomous init.
     driverController.back().onTrue(new ArmCalibrateCMD(arm, led2023));
+    driverController.a().onTrue(new ArmCalibrateZeroAtHomeCMD(arm, led2023));
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/RobotContainer.java:317]: This should be changed to a different button

File: src/main/java/frc/robot/subsystems/arm/Arm.java [hunk 2]
```diff
@@ -265,7 +275,8 @@ private void autoPeriodic() {
     logger.recordOutput("Arm/AutoMode", autoMode.toString());
     switch (autoMode) {
       case RETRACT:
-        if (armIOInputs.extendPosition < SAFE_RETRACT_NON_HOME + EXTEND_TOLERANCE_METERS) {
+        if (autoRetractTimer.hasElapsed(1)
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/subsystems/arm/Arm.java:278]: Does this timer impl reset and start? Timers should be used in either periodic or a command

**Review by AdamCraftmaster — CHANGES_REQUESTED** [review]
Needs a button binding change and a quick check on the timer, otherwise lgtm

File: src/main/java/frc/robot/subsystems/arm/Arm.java (hunk 2 above)
Comment (elleryjh) [reply src/main/java/frc/robot/subsystems/arm/Arm.java:278]: Yes, it has reset() and start() in setTargetPositions.

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (elleryjh) [reply src/main/java/frc/robot/RobotContainer.java]: Moved to B.

**Review by AdamCraftmaster — APPROVED** [review]
lgtm

**Pull request merged**